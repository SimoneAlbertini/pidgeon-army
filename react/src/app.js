import React from 'react'
import Hello from './hello';

global.Hello = Hello;