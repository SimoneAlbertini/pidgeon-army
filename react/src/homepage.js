import React from "react";

class AddContactForm extends React.Component {
    render_field(name, label) {
        let nameprop = 'add_form-' + name;
        return (
            <div>
                <label for={nameprop}>{label}</label>
                <input type="text" name={nameprop}/>
            </div>
        )
    }

    render() {
        return (
            <form id="add_contact_form" action="#" className="addcontactform" method="POST">
                {this.render_field('name', 'First name:')}
                {this.render_field('lastname', 'Last name:')}
                {this.render_field('email', 'Email:')}
                <button type="submit">Add</button>
            </form>
        )
    }
}

class ContactsTable extends React.Component {
    render() {
        return (
            <table id="contacts_table" className="contactstable">
                <tr>
                    <td>&nbsp;</td>
                    <td>&nbsp;</td>
                </tr>
            </table>
        )
    }
}

export default class ManageContacts extends React.Component {
    render() {
        return (
            <div>
                <AddContactForm/>
                <ContactsTable/>
            </div>
        );
    }
}

