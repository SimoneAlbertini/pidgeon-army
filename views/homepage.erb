<h1>Gestione contatti</h1>
<div id="root"></div>