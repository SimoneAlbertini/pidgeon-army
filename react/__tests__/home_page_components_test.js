import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {AddContactForm, ContactsTable} from '../src/homepage';

describe('<AddContactForm />', () => {
    test('snapshot', () => {
        const component = renderer.create(<AddContactForm/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('has add button', () => {
        const form = shallow(<AddContactForm/>);
        expect(form.find('button').text()).toEqual('Add');
    });
});

describe('<ContactsTable />', () => {
    it('calls componentDidMount', () => {
        const spy = jest.spyOn(ContactsTable.prototype, 'componentDidMount');
        const wrapper = mount(<ContactsTable />);

        expect(spy).toHaveBeenCalled();
    });

    it('renders fetched data in rows', () => {
        fail('TODO');
    });
});