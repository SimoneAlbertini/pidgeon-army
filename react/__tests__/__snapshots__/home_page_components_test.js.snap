// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddContactForm /> snapshot 1`] = `
<form
  action="#"
  id="add_contact_form"
  method="POST"
>
  <div>
    <label
      htmlFor="add_form-name"
    >
      First name:
    </label>
    <input
      name="add_form-name"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="add_form-lastname"
    >
      Last name:
    </label>
    <input
      name="add_form-lastname"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="add_form-email"
    >
      Email:
    </label>
    <input
      name="add_form-email"
      type="text"
    />
  </div>
  <button
    type="submit"
  >
    Add
  </button>
</form>
`;
